src/samples/OpenChannelMessageThreading.js

```diff
@@ -0,0 +1,718 @@
+      updateState({ ...state, messages: messages, messageInputValue: "", messageToUpdate: null });
+    } else {
+      const userMessageParams = new UserMessageCreateParams();
+
+      userMessagesHandler(false, userMessageParams, messages)
+    }
+  }
+
+  const sendThreadMessage = () => {
+    const { threadMessages, threadParentsMessage } = state;
+    const userMessageParams = new UserMessageCreateParams({ parentMessageId: threadParentsMessage.messageId });
+
+    userMessagesHandler(true, userMessageParams, threadMessages)
+  }
+
+  const fileMessagesHandler = (fileMessageParams, messages, isThread, event) => {
+    const { currentlyJoinedChannel } = state;
+    fileMessageParams.file = event.currentTarget.files[0];
+
+    currentlyJoinedChannel.sendFileMessage(fileMessageParams).onSucceeded((message) => {
+      const updatedMessages = [...messages, message];
+      updateState(() => {
+        if (isThread) {
+          return { ...state, threadMessages: updatedMessages, threadMessageInputValue: "", threadFile: null }
+        }
+
+        return { ...state, messages: updatedMessages, messageInputValue: "", file: null }
+      });
+
+    }).onFailed((error) => {
+      console.log(error)
+      console.log("failed")
+    });
+  }
+
+  const onFileInputChange = async (e) => {
+    if (e.currentTarget.files && e.currentTarget.files.length > 0) {
+      const { messages } = state;
+      const fileMessageParams = new FileMessageCreateParams();
+
+      fileMessagesHandler(fileMessageParams, messages, false, e);
+    }
+  }
+
+  const onFileThreadInputChange = async (e) => {
+    if (e.currentTarget.files && e.currentTarget.files.length > 0) {
+      const { threadMessages, threadParentsMessage } = state;
+      const fileMessageParams = new FileMessageCreateParams({parentMessageId: threadParentsMessage.messageId});
+
+      fileMessagesHandler(fileMessageParams, threadMessages, true, e);
+    }
+  }
+
+  const handleDeleteMessage = async (messageToDelete) => {
+    const { currentlyJoinedChannel } = state;
+    await deleteMessage(currentlyJoinedChannel, messageToDelete); // Delete
+
+  }
+
+  const updateMessage = async (message) => {
+    updateState({ ...state, messageToUpdate: message, messageInputValue: message.message });
+  }
+
+  const openThread = async (parentsMessage) => {
+    const { currentlyJoinedChannel } = state;
+
+    const params = new MessageRetrievalParams({
+      messageId: parentsMessage.messageId,
+      channelType: "open", // Acceptable values are open and group.
+      channelUrl: currentlyJoinedChannel.url,
+    });
+
+    const paramsThreadedMessageListParams = new ThreadedMessageListParams({
+      prevResultSize: 10,
+      nextResultSize: 10,
+      isInclusive: true,
+      reverse: false,
+      includeParentMessageInfo: false,
+    })
+
+    const { threadedMessages } = await parentsMessage.getThreadedMessagesByTimestamp(30, paramsThreadedMessageListParams);
```

Comment:
Can this go into its own function below. A function which also includes the param construction on the lines above